### ISSUE TITLE
[TF Core] Support the use of the third-party KDNN library

### ISSUE THREAD

**Issue by amaniyadunia** [tensorflow/tensorflow#124076]
### Issue type

Feature Request

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

v2.15.0

### Custom code

Yes

### OS platform and distribution

openEuler 24.03

### Python version

3.11

### Current behavior?

KDNN provides high-performance implementations of a series of functions. We expect TensorFlow to be able to call KDNN to improve the execution performance of certain computational graphs.
Currently, we have completed the replacement and verification of the sigmoid operator based on version 2.15.0. This is the relevant code: https://github.com/amaniyadunia/tensorflow/commit/7f28db941e4d4a7bf3fa3cfad5e0244bcef4a677.
We have implemented a custom operator that calls the KDNN function and replaces it during the graph compilation stage using the remapper mechanism, thereby achieving the goal of improving performance by using the KDNN algorithm library.
Subsequently, plans are also in place to complete the integration of Matmul, floormod, Softmax, concat, and eisum operators.

### Standalone code to reproduce the issue

```shell
Code reference for the custom remapper and sigmoid operator implementation:
https://github.com/amaniyadunia/tensorflow/commit/7f28db9
```

### Relevant log output

```shell

```
